python/mlc_llm/op/top_p_pivot.py

```diff
@@ -4,11 +4,34 @@
 from tvm.script import tir as T
 
 # mypy: disable-error-code="attr-defined,valid-type,name-defined"
-# pylint: disable=too-many-locals,invalid-name,too-many-arguments,
+# pylint: disable=too-many-locals,invalid-name,too-many-arguments,unnecessary-lambda
 # pylint: disable=too-many-statements,line-too-long,too-many-nested-blocks,too-many-branches
 
 
 def top_p_pivot(pN):
+    """Top-p pivot function. This function finds the pivot to cut-off top-p percentile.
+
+    A valide pivot should satisfy the following conditions:
+    - lsum >= top_p
+    - top_p > lsum - cmin * lmin
+    where lsum is the sum of elements that are larger or equal to the pivot,
+    lmin is the minimum elements that is larger or equal to the pivot,
+    cmin is the count of elements that are equal to lmin,
+
+    Parameters
+    ----------
+    prob:
+        The probability vector
+
+    top_p_global:
+        The top-p threshold
+
+    init_pivots:
+        The initial pivot candidates
+
+    final_pivot:
```

Comment:
we should followup with a renorm kernel, and also return the lsum(perhaps together with final pivot in one array or another)